### PR TITLE
SMT2 backend testing

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 1db210ebb85f117c1f5127c479cdd38375a7d192).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89387778

File: scripts/delete_failing_smt2_solver_tests [hunk 1]
```diff
@@ -0,0 +1,315 @@
+#!/bin/sh
+
+cd regression/cbmc
+rm ACSL/operators.desc
```
Comment (peterschrammel) [inline scripts/delete_failing_smt2_solver_tests:4]: Would using something like `-X cprover-smt-solver` be the more appropriate solution here?

File: src/cbmc/cbmc_parse_options.h [hunk 2]
```diff
@@ -46,7 +46,8 @@ class optionst;
   OPT_GOTO_CHECK \
   "(no-assertions)(no-assumptions)" \
   "(xml-ui)(xml-interface)(json-ui)" \
-  "(smt1)(smt2)(fpa)(cvc3)(cvc4)(boolector)(yices)(z3)(opensmt)(mathsat)" \
+  "(smt1)(smt2)(fpa)" \
+  "(cprover-smt2)(cvc3)(cvc4)(boolector)(yices)(z3)(opensmt)(mathsat)" \
```
Comment (peterschrammel) [inline src/cbmc/cbmc_parse_options.h]: Put on separate line to facilitate merging

File: src/solvers/smt2/smt2_conv.cpp [hunk 3]
```diff
@@ -72,6 +72,10 @@ void smt2_convt::write_header()
   {
   case solvert::GENERIC: break;
   case solvert::BOOLECTOR: out << "; Generated for Boolector\n"; break;
+  // clang-format off
```
Comment (peterschrammel) [inline src/solvers/smt2/smt2_conv.cpp]: This should be put around the entire switch if we really don't want to reformat it.

File: scripts/delete_failing_smt2_solver_tests (hunk 1 above)
Comment (kroening) [reply scripts/delete_failing_smt2_solver_tests:4]: I hope we'll get there eventually -- however, at the moment, this would touch >300 files!

**Review by tautschnig — APPROVED** [review]
Having some SMT solver run in CI is a step forward, just the formatting comments that @peterschrammel remarked should be addressed I'd say.

File: src/cbmc/cbmc_parse_options.h (hunk 2 above)
Comment (kroening) [reply src/cbmc/cbmc_parse_options.h]: done

File: src/solvers/smt2/smt2_conv.cpp (hunk 3 above)
Comment (kroening) [reply src/solvers/smt2/smt2_conv.cpp]: done

**Review by martin-cs — APPROVED** [review]
Seems like progress.

**Comment by tautschnig** [comment]
@kroening clang-format isn't happy yet.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 04954727b932eb73f0d2a4c37b1d0885411054bc).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89644799
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Comment by thk123** [comment]
Could this be rebased - I think this is the cause of the TG failure.

**Comment by tautschnig** [comment]
~How can a PR that isn't even merged be causing TG failures?~

**Comment by tautschnig** [comment]
I'm sorry, please ignore my previous post, I suppose you were saying that rebasing this PR will make the bot happy for this PR.

**Comment by kroening** [comment]
Rebased

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 43835116040f99c988970d5b10973ab04641492f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90271880

**Review by forejtv — APPROVED** [review]

**Pull request merged**